### PR TITLE
Implemented cmd: template repository init

### PR DESCRIPTION
**Pull request by webrouse**
Part of: https://keboola.atlassian.net/browse/COM-1210

### CODE REVIEW

File: internal/pkg/cli/dependencies/container.go [hunk 1]
```diff
@@ -287,15 +290,37 @@ func (c *Container) CreateManifest(o createManifest.Options) (*manifest.Manifest
...
+			c.projectManifest = m
 		} else {
 			return nil, err
 		}
 	}
-	return c.manifest, nil
+	return c.projectManifest, nil
+}
+
+func (c *Container) CreateRepositoryManifest() (*repositoryManifest.Manifest, error) {
```
Comment (webrouse) [inline internal/pkg/cli/dependencies/container.go:304]: Pridany `RepositoryManifest` do `dependencies.Container`.

File: internal/pkg/cli/dependencies/container.go [hunk 2]
```diff
@@ -287,15 +290,37 @@ func (c *Container) CreateManifest(o createManifest.Options) (*manifest.Manifest
 	}
 }
 
-func (c *Container) Manifest() (*manifest.Manifest, error) {
-	if c.manifest == nil {
-		if m, err := loadManifest.Run(c); err == nil {
-			c.manifest = m
+func (c *Container) ProjectManifest() (*projectManifest.Manifest, error) {
```
Comment (webrouse) [inline internal/pkg/cli/dependencies/container.go:293]: `Manifest` -> `ProjectManifest`, kedze uz mame 2 druhy manifestov, project a repository.

File: internal/pkg/local/manager_test.go [hunk 3]
```diff
@@ -169,7 +169,7 @@ func TestLocalLoadMapper(t *testing.T) {
 	testhelper.ReplaceEnvsDir(fs, `/`, envs)
 
 	// Load objects
-	m, err := manifest.LoadManifest(fs, zap.NewNop().Sugar())
+	m, err := manifest.Load(fs, zap.NewNop().Sugar())
```
Comment (webrouse) [inline internal/pkg/local/manager_test.go:172]: Blby naming, boli metody `Save` a `LoadManifest` -> `Load`.

File: internal/pkg/cli/cmd/template/repository/init.go [hunk 4]
```diff
@@ -22,7 +21,7 @@ func InitCommand(depsProvider dependencies.Provider) *cobra.Command {
 				return err
 			}
 
-			return fmt.Errorf(`not implemented`)
+			return initOp.Run(d)
```
Comment (webrouse) [inline internal/pkg/cli/cmd/template/repository/init.go:24]: Implementovany prikaz `template repository init`.

Comment (webrouse) [inline internal/pkg/cli/cmd/template/repository/init.go:24]: Implementovany prikaz `template repository init`.

File: internal/pkg/template/repository/manifest/manifest.go [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+const FileName = `repository.json`
+
+type Manifest struct {
+	fs       filesystem.Fs
+	changed  bool
+	*Content `validate:"required,dive"` // content of the file, updated only on load/save
+}
+
+type Content struct {
+	Version int `json:"version" validate:"required,min=1,max=2"`
```
Comment (webrouse) [inline internal/pkg/template/repository/manifest/manifest.go:24]: `RepositoryManifest` obsahuje zatial iba verziu, ... dalsie `fields` budu pridavane v dalsich PRs.

File: pkg/lib/operation/template/local/manifest/create/operation.go [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+	"github.com/keboola/keboola-as-code/internal/pkg/filesystem"
+	"github.com/keboola/keboola-as-code/internal/pkg/template/repository/manifest"
+)
+
+type dependencies interface {
+	Logger() *zap.SugaredLogger
+	EmptyDir() (filesystem.Fs, error)
+}
+
+func Run(d dependencies) (*manifest.Manifest, error) {
```
Comment (webrouse) [inline pkg/lib/operation/template/local/manifest/create/operation.go:15]: Operation: vytvorenie repository manifest

File: pkg/lib/operation/template/local/repository/init/operation.go [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+)
+
+type dependencies interface {
+	Ctx() context.Context
+	Logger() *zap.SugaredLogger
+	EmptyDir() (filesystem.Fs, error)
+	CreateRepositoryManifest() (*manifest.Manifest, error)
+}
+
+func Run(d dependencies) (err error) {
```
Comment (webrouse) [inline pkg/lib/operation/template/local/repository/init/operation.go:20]: Operation: init repository

File: internal/pkg/build/build.go [hunk 8]
```diff
@@ -1,7 +1,10 @@
 //nolint:gochecknoglobals
 package build
 
-const DevVersionValue = "dev"
+const (
+	DevVersionValue = "dev"
+	MajorVersion    = 2
```
Comment (webrouse) [inline internal/pkg/build/build.go:6]: Pouziva sa v project a repository manifest, aktualna verzia, ... ak by sme v buducnosti potrebovali migrovat.

File: internal/pkg/template/repository/manifest/manifest_test.go [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+
+func minimalStruct() *Content {
+	return &Content{
+		Version: 2,
+	}
+}
+
+func fullJson() string {
+	return `{
+  "version": 2
```
Comment (webrouse) [inline internal/pkg/template/repository/manifest/manifest_test.go:140]: Bude doplnene neskor od dalsie fields.

File: test/repo-init/out/.keboola/repository.json [hunk 10]
```diff
@@ -0,0 +1,3 @@
+{
+  "version": 2
```
Comment (webrouse) [inline test/repo-init/out/.keboola/repository.json:2]: Funkcionalny test: uspesna inicializacia.

File: test/repo-init-not-empty-dir/expected-stderr [hunk 11]
```diff
@@ -0,0 +1,4 @@
+Error:
+  - directory "%s" it not empty, found:
```
Comment (webrouse) [inline test/repo-init-not-empty-dir/expected-stderr:2]: Funkcionalny test: adresar nie je prazdny.

**Review by JakubMatejka — APPROVED** [review]

**Pull request merged**